### PR TITLE
Gra 991 local connection

### PR DESCRIPTION
**Pull request by mattkasun**
sets Default Interface on join:

tested on linux/freebsd/darwin  ; window status unknown

### CODE REVIEW

File: functions/localport_freebsd.go [hunk 1]
```diff
@@ -1,82 +1,50 @@
-//go:build freebsd
-// +build freebsd
+//go:build freebsd || darwin
```
Comment (afeiszli) [inline functions/localport_freebsd.go:1]: should the dawrin functions be in the localport_freebsd.go file? or maybe it should be renamed

**Review by afeiszli — COMMENTED** [review]
was anything written for finding the default interface on windows?

File: functions/localport_freebsd.go (hunk 1 above)
Comment (0xdcarns) [reply functions/localport_freebsd.go:1]: sgould be _unix with !linux

File: functions/localport_linux.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package functions
+
+import (
+	"errors"
+	"net"
+
+	"github.com/vishvananda/netlink"
+)
+
+func getDefaultInterface() (*string, error) {
```
Comment (0xdcarns) [inline functions/localport_linux.go:10]: what is point of returning reference to string?

File: functions/join.go [hunk 3]
```diff
@@ -279,6 +279,13 @@ func JoinNetwork(flags *viper.Viper) (*config.Node, *config.Server, error) {
 			host.Interfaces = *ip
 		}
 	}
+	defaultInterface, err := getDefaultInterface()
+	if err != nil {
+		logger.Log(0, "default gateway not found", err.Error())
+	} else {
+		host.DefaultInterface = *defaultInterface
```
Comment (0xdcarns) [inline functions/join.go:286]: see below comment if this just getting de-referenced anyways

**Review by 0xdcarns — CHANGES_REQUESTED** [review]

**Review by 0xdcarns — APPROVED** [review]

**Review by theguy951357 — COMMENTED** [review]
can't get it built on windows. I did go get github.com/gravitl/netmaker@develop, then go mod tidy. wails build -skipbindings shows:
functions\join.go:282:27: undefined: getDefaultInterface

**Review by abhishek9686 — APPROVED** [review]

**Comment by afeiszli** [comment]
Please add an unimplemented function for Windows

**Comment by theguy951357** [comment]
got this on windows builld 

 github.com/gravitl/netclient/functions
functions\proxy.go:21:22: not enough arguments in call to setupMQTTSingleton
        have (*"github.com/gravitl/netclient/config".Server)
        want (*"github.com/gravitl/netclient/config".Server, bool)

**Review by theguy951357 — APPROVED** [review]
windows builds now
pass

**Pull request merged**